workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/GetAppPage.tsx

```diff
@@ -52,83 +75,114 @@ const GetAppFlow = () => {
+  async function handleGetApp() {
+    const productSpecificationValues = await getProductSpecificationValues(
+      productId
+    );
+
+    const orderType = await getProductOrderTypes(productSpecificationValues);
+    const cart = buildNewCart(
+      billingAddress,
+      channel,
+      email,
+      isFreeApp,
+      orderType,
+      product,
+      purchaseOrderNumber,
+      selectedAccount,
+      selectedPaymentMethod,
+      sku
+    );
+
+    const cartResponse = await postCartByPaymentMethod(cart, channel.id);
+
+    await postCheckoutCart({ cartId: cartResponse.id });
+
+    const nextStepsCallbackURL = `${Liferay.ThemeDisplay.getCanonicalURL().replace(
+      "/get-app",
+      ""
+    )}/next-steps?orderId=${encodeURIComponent(cartResponse.id)}`;
+
+    if (selectedPaymentMethod === "pay") {
+      const paymentMethodURL = await getPaymentMethodURL(
+        cartResponse.id,
+        nextStepsCallbackURL
+      );
+
+      window.location.href = paymentMethodURL;
+    } else {
+      window.location.href = nextStepsCallbackURL;
+    }
+  }
+
+  const StepFormComponent: StepComponent = {
+    [StepType.ACCOUNT]: (
+      <AccountSelection
+        onSelectAccount={(account: Account) => {
+          setValue("selectedAccount", account);
+          setShowAccount(true);
+        }}
+        setUserAccount={setUserAccount}
+        userAccount={userAccount}
+      />
+    ),
+    [StepType.PAYMENT]: (
+      <SelectPaymentMethod
+        addresses={addresses}
+        billingAddress={billingAddress}
+        email={email}
+        enableTrialMethod={enableTrialMethod}
+        purchaseOrderNumber={purchaseOrderNumber}
+        selectedPaymentMethod={selectedPaymentMethod}
+        setBillingAddress={setBillingAddress}
+        setEmail={setEmail}
+        setEnablePurchaseButton={setEnablePurchaseButton}
+        setPurchaseOrderNumber={setPurchaseOrderNumber}
+        setSelectedPaymentMethod={setSelectedPaymentMethod}
+      />
+    ),
+  };
+
+  return (
+    <>
+      <ProductCard
+        productId={Number(getUrlParam("productId"))}
+        selectedAccount={getValues("selectedAccount")}
+        setProductToForm={(product: Product) => setValue("product", product)}
+        showAccount={showAccount}
+      ></ProductCard>
+
+      <div className="border d-flex flex-column mt-7 p-5 rounded">
+        <div className="d-flex flex-column">
+          <div className="align-self-center h1 mb-6">
+            {sectionProperties[step].title}
+          </div>
+          <div>{StepFormComponent[step]}</div>
+        </div>
+        <ProductFooter
+          addresses={addresses}
+          enablePurchaseButton={enablePurchaseButton}
+          handleGetApp={handleGetApp}
+          isFreeApp={isFreeApp}
+          sectionProperties={sectionProperties}
+          selectedAccount={selectedAccount}
+          selectedPaymentMethod={selectedPaymentMethod}
+          setStep={setStep}
+          sku={sku}
+          step={step}
+        />
```

Comment:
```suggestion
        <ProductFooter
            {addresses, 
            enablePurchaseButton, 
            handleGetApp, 
            isFreeApp, 
            sectionProperties,
            selectedAccount, 
            selectedPaymentMethod, 
            setStep, 
            sku, 
            step}
        />
```
I think it may work.

Reply:
This didn't work :(